Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -1372,8 +1374,11 @@ def visit_WithStatNode(self, node):
                     name, value = directive
                     if name in ('nogil', 'gil'):
                         # special case: in pure mode, "with nogil" spells "with cython.nogil"
-                        node = Nodes.GILStatNode(node.pos, state = name, body = node.body)
+                        condition = node.manager.args[0] if isinstance(node.manager, ExprNodes.CallNode) else None
+                        node = Nodes.GILStatNode(node.pos, state = name, body = node.body, condition=condition)
                         return self.visit_Node(node)
+                    elif isinstance(value, PostParseError):
+                        raise value
                     if self.check_directive_scope(node.pos, name, 'with statement'):
                         directive_dict[name] = value
```

Comment:
Now, I think that there can be a problem here when `value` is PostParseError exception...